### PR TITLE
8255213: Shenandoah build fails without JFR after JDK-8255036

### PR DESCRIPTION
**Pull request by DamonFool**
Hi all,

Please review the one-line fix of Shenandoah build failure without JFR.

Thanks.
Best regards,
Jie

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Testing

|     | Linux x32 | Linux x64 | Windows x64 | macOS x64 |
| --- | ----- | ----- | ----- | ----- |
| Build | ✔️ (1/1 passed) | ✔️ (5/5 passed) | ✔️ (2/2 passed) | ✔️ (2/2 passed) |
| Test (tier1) |    |  ✔️ (9/9 passed) | ❌ (1/9 failed) | ✔️ (9/9 passed) |

**Failed test task**
- [Windows x64 (hs/tier1 compiler)](https://github.com/DamonFool/jdk/runs/1291363354)

### Issue
 * [JDK-8255213](https://bugs.openjdk.java.net/browse/JDK-8255213): Shenandoah build fails without JFR after JDK-8255036

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/794/head:pull/794`
`$ git checkout pull/794`

### CODE REVIEW

File: src/hotspot/share/gc/shenandoah/shenandoahRootVerifier.cpp
Comment (shipilev) [inline src/hotspot/share/gc/shenandoah/shenandoahRootVerifier.cpp:33]: I think you can drop the `#include "gc/shenandoah/shenandoahHeap.hpp"` then, as `.inline.hpp` is supposed to transitively include it.

**Review by shipilev — CHANGES_REQUESTED** [review]

File: src/hotspot/share/gc/shenandoah/shenandoahRootVerifier.cpp
Comment (DamonFool) [reply src/hotspot/share/gc/shenandoah/shenandoahRootVerifier.cpp:32]: Good idea. 
Updated.
Thanks.

**Review by shipilev — APPROVED** [review]
Looks fine, assuming nopch builds come clean. (They should, but still wait for them?)

**Comment by DamonFool** [comment]
@shipilev 

One more build error occurred if nopch and -jfr.
```
/home/jvm/jiefu/open/jdk/src/hotspot/share/gc/shenandoah/c2/shenandoahBarrierSetC2.cpp: In member function 'void ShenandoahBarrierSetC2::insert_pre_barrier(GraphKit*, Node*, Node*, Node*, bool) const':
/home/jvm/jiefu/open/jdk/src/hotspot/share/gc/shenandoah/c2/shenandoahBarrierSetC2.cpp:376:27: error: 'java_lang_ref_Reference' has not been declared
       otype->get_con() != java_lang_ref_Reference::referent_offset()) {
                           ^~~~~~~~~~~~~~~~~~~~~~~
/home/jvm/jiefu/open/jdk/src/hotspot/share/gc/shenandoah/c2/shenandoahBarrierSetC2.cpp:416:32: error: 'java_lang_ref_Reference' has not been declared
   Node* referent_off = __ ConX(java_lang_ref_Reference::referent_offset());
                                ^~~~~~~~~~~~~~~~~~~~~~~
```

I've updated the patch to fix it.
Could you please review it again?
Thanks.

**Review by shipilev — APPROVED** [review]

**Comment by DamonFool** [comment]
Hi @shipilev ,
Do you think this change is trivial?
Or shall we wait for one more review?
Thanks.

**Comment by shipilev** [comment]
I believe you can integrate.

**Pull request closed without merging**
